app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -35,6 +37,42 @@ def events
       respond_for_index :collection => collection
     end
 
+    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
+    param :name, String, :desc => N_("Name of the content host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
+    param :facts, Hash, :desc => N_("Key-value hash of content host-specific facts") do
+      param :fact, String, :desc => N_("Any number of facts about this content host")
+    end
+    param :type, String, :desc => N_("Type of the content host, it should always be 'system'")
+    param :guest_ids, Array, :desc => N_("IDs of the virtual guests running on this content host")
+    param :installed_products, Array, :desc => N_("List of products installed on the content host")
```

Comment:
Currently, these are passed straight through to candlepin and need to be camel case
```
[{productId: "MCT00123", productName: "Some Red Hat product"}]
```

These products don't necessarily exist in katello db. Accepting snake-case and converting would be beneficial.